cassandra-spark/src/main/scala/geotrellis/spark/store/cassandra/CassandraRDDReader.scala

```diff
@@ -63,35 +61,35 @@ object CassandraRDDReader {
 
     val bins = IndexRanges.bin(ranges, numPartitions.getOrElse(sc.defaultParallelism))
 
-    val query = QueryBuilder.select("value")
-      .from(keyspace, table)
-      .where(eqs("key", QueryBuilder.bindMarker()))
-      .and(eqs("name", layerId.name))
-      .and(eqs("zoom", layerId.zoom))
-      .toString
+    val query = QueryBuilder
+      .selectFrom(keyspace, table)
+      .column("value")
+      .whereColumn("key").isEqualTo(QueryBuilder.bindMarker())
+      .whereColumn("name").isEqualTo(literal(layerId.name))
+      .whereColumn("zoom").isEqualTo(literal(layerId.zoom))
+      .asCql()
 
     sc.parallelize(bins, bins.size)
       .mapPartitions { partition: Iterator[Seq[(BigInt, BigInt)]] =>
         instance.withSession { session =>
-          implicit val ec = executionContext
+          implicit val ioRuntime: unsafe.IORuntime = runtime
           val statement = session.prepare(query)
 
           val result = partition map { seq =>
-            IOUtils.parJoin[K, V](seq.iterator) { index: BigInt =>
-              val row = session.execute(statement.bind(index: BigInteger))
-              if (row.asScala.nonEmpty) {
-                val bytes = row.one().getBytes("value").array()
-                val recs = AvroEncoder.fromBinary(kwWriterSchema.value.getOrElse(_recordCodec.schema), bytes)(_recordCodec)
-                if (filterIndexOnly) recs
-                else recs.filter { row => includeKey(row._1) }
-              } else Vector.empty
+            IOUtils.parJoinIO[K, V](seq.iterator) { index: BigInt =>
+              session.executeF[IO](statement.bind(index.asJava)).map { row =>
```

Comment:
It's possible to implement it in a similar fashion for S3, see https://github.com/locationtech/geotrellis/issues/3467